### PR TITLE
Issue #61 skip option

### PR DESCRIPTION
**Pull request by adammcnabb**
add option to skip dependency-track maven plugin execution based on configuration value <skip> or CLI argument `dependency-track.skip`.  

Based on the approach used by SonarQube's Maven Scanner (https://github.com/SonarSource/sonar-scanner-maven/blob/master/src/main/java/org/sonarsource/scanner/maven/SonarQubeMojo.java).  

- Added unit tests to verify no interactions resulting from an execution of the Mojo objects. 
- refactored the test resources directories to provide a separate pom.xml that is `<skip>true</skip>`.  
- made constant and static utility classes final and private constructors to avoid instantiation or extension.

### CODE REVIEW

File: src/test/resources/projects/skip/bom.xml [hunk 1]
```diff
@@ -0,0 +1,3204 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adammcnabb) [inline src/test/resources/projects/skip/bom.xml:1]: Straight copy from the projects/run/bom.xml.  no differences.

**Comment by adammcnabb** [comment]
@pmckeown would love to get this merged and released for use by my org.  when you have time to review, much appreciated.  happy to discuss if there is anything you would prefer to change.

**Comment by pmckeownkiwibank** [comment]
> @pmckeown would love to get this merged and released for use by my org. when you have time to review, much appreciated. happy to discuss if there is anything you would prefer to change.

Hey sorry had a busy week and was unable to complete testing on the weekend.  Will get onto this tonight.

File: src/test/resources/projects/skip/bom.xml (hunk 1 above)
Comment (pmckeown) [reply src/test/resources/projects/skip/bom.xml:1]: This file isn't needed.  I'd rather it was included in the merge as it will just need to be removed.

**Review by pmckeown — CHANGES_REQUESTED** [review]
Looks good - just a couple of minor points on unnecessary stubbing (though I suspect there is a bit of that going on) so I would have been happy to merge them.  

However the new skip/bom.xml file isn't actually needed to run a clean install so I'd rather it wasn't included in the merge.

**Review by pmckeown — APPROVED** [review]
I'll actually get this merged and tidied myself as I won't get another chance prior to Christmas!

**Comment by pmckeown** [comment]
Thanks for adding this @adammcnabb :)

Its been released to the Sonatype OSS repo - should be in maven central over night.

https://oss.sonatype.org/service/local/repositories/releases/content/io/github/pmckeown/dependency-track-maven-plugin/1.1.0/dependency-track-maven-plugin-1.1.0.jar

File: src/test/java/io/github/pmckeown/dependencytrack/metrics/MetricsMojoIntegrationTest.java [hunk 2]
```diff
@@ -119,4 +117,17 @@ public void thatAnyCriticalIssuesPresentCanFailTheBuild() throws Exception {
             assertThat(ex, is(instanceOf(MojoFailureException.class)));
         }
     }
+
+    @Test
+    public void thatTheMetricsIsSkippedWhenSkipIsTrue() throws Exception {
+        stubFor(get(urlEqualTo(V1_PROJECT)).willReturn(
+                aResponse().withBodyFile("api/v1/project/get-all-projects.json")));
```
[the remark this answers is not in the document]
Comment (adammcnabb) [reply src/test/java/io/github/pmckeown/dependencytrack/metrics/MetricsMojoIntegrationTest.java:124]: If the skip failed, I wanted to ensure that the stubbing was in place to provide a response for verification.  Unsure if that was best practice with Wiremock however.

**Pull request merged**